### PR TITLE
Add tests for optional value dictionaries

### PR DESCRIPTION
**Pull request by mnot**

### CODE REVIEW

**Comment by mnot** [comment]
@phluid61?

**Comment by phluid61** [comment]
Seems alright. Do we need to test for a dictionary where all members are true (identical to a list)? Do we want to have a case with an illegal `x=?1` with a proper canonical form?

**Comment by mnot** [comment]
Done, thanks.

File: dictionary.json [hunk 1]
```diff
@@ -103,6 +109,13 @@
       "header_type": "dictionary",
       "expected": {"a": [1,{}], "b": [true, {"foo": 9}], "c": [3, {}]}
     },
+    {
+        "name": "missing value with params dictionary, irregular serialisation",
+        "raw": ["a=1, b;foo=9, c=3"],
```
Comment (phluid61) [inline dictionary.json]: This isn't allowed by either the ABNF or the algorithm. Parameters are part of the member value, but the member value is omitted in this serialisation.

Comment (phluid61) [reply dictionary.json]: Logged as [#1007](https://github.com/httpwg/http-extensions/issues/1007) in case the test is correct, and the spec has a bug.

**Pull request merged**